elintegrostartapp/grails-app/controllers/com/elintegro/translator/TranslatorAssistantController.groovy

```diff
@@ -30,9 +30,13 @@ class TranslatorAssistantController {
         String projectName = project.name
         String language = project.sourceLanguage
         String fileName = project.sourceFile
-        def result = new FileUploadController().ajaxFileSaveWithParams(params, projectName)
-        translatorService.loadTexts(fileName, language, projectName)
-        render(success: true)
+        def validateEnglish = translatorService.validateSourceFile(params,projectName,fileName)
+        if (validateEnglish == true) {
+            def result = new FileUploadController().ajaxFileSaveWithParams(params, projectName)
```

Comment:
FileUploadController is not a good place to put validation. Please move this method to FileValidationService.groovy 
You can inject the service in your TranslatorAssistantController and use it. If you need the same method in other controller (like FileValidationController) just inject the service there and use it.